UHERO/udaman
# Add universe to the universe (UA-652)

**NOTE:** This branch is rooted at UA-662, so you must diff against that branch. Probably not the best idea in hindsight, but it seemed "like a good idea at the time" :=\ When this PR is approved, let's go ahead and promote both of these to production. That will leave UA-512 (which is currently on hold) behind (based on my original plan), but I will deal with back-merging master into it and dealing with conflicts when the time comes. Shouldn't be too bad.
As long as this PR goes through as-is, I will expect to put in a backlog story to convert the many hard-coded instances of string "UHERO" to something like `ENV['DEFAULT_UNIV']` which would be defined in the config, and perhaps each resource model class should have a method which returns `Model.where(universe: ENV['DEFAULT_UNIV'])`, to simplify the code further.
### Testing
First, run the following SQL to populate the universe column (this is also given in JIRA). Note that in most cases there's no need to explicitly set universe = UHERO, because it's the column default.
```
update categories set universe = 'UHERO' where universe is null;
update data_lists set universe = 'DBEDT' where name like 'DBEDT%';
update data_sources set universe = 'DBEDT' where eval like '%Dbedt%';
update measurements set universe = 'DBEDT' where prefix like 'DBEDT%';
update series set universe = 'DBEDT' where name like 'DBEDT%';
update sources set universe = 'DBEDT' where source_type = 'DBEDT';
update users set universe = 'DBEDT' where role = 'dbedt';
update data_points d join series s on d.series_id = s.id
   set d.universe = s.universe where d.universe <> s.universe or d.universe is null;
update public_data_points d join series s on d.series_id = s.id
   set d.universe = s.universe where d.universe <> s.universe or d.universe is null;
```
* Check index pages of resources like Measurements and Data Lists to see that they show only UHERO and not DBEDT items. In particular Sources index page currently in prod shows both DBEDT and UHERO sources, so this can be compared for difference
* Check specialty pages like `/series/no_source` and `/series/quarantine` for sensible output
* Check the autocomplete in the series search box
* Do a test of DBEDT loading and changing active upload (hassle :=P)
* As long as you've got redis and sidekiq running, do a rails console test of `Series.reload_by_dependency_depth`. I passed in a small subset of series just to save time. My test command: `Series.reload_by_dependency_depth(Series.where('universe = "UHERO" AND name like "GC%"'))`
